blocks/carousel/carousel.js

```diff
@@ -0,0 +1,452 @@
+  const currentActiveDot = carouselSlider.querySelector('.dot.active');
+  if (currentActiveDot && currentActiveDot.dataset.index === activeDot.dataset.index) {
+    return;
+  }
+  const carouselTrack = carouselSlider.querySelector('.carousel-track');
+  const widthAvailable = carouselTrack.offsetWidth;
+  const allowedCards = allowedCardsCount();
+  const cardWidth = widthAvailable / allowedCards;
+  const cards = Array.from(carouselTrack.children);
+  cards.forEach((card, index) => {
+    if (index >= dotIndex && index < dotIndex + allowedCards) {
+      card.style.opacity = 1;
+    } else {
+      card.style.opacity = 0;
+    }
+    card.style.width = `${cardWidth}px`;
+  });
+  const moveDistance = dotIndex * cards[0].offsetWidth;
+  carouselTrack.style.transform = `translateX(-${moveDistance}px)`;
+  dots.forEach((dot) => dot.classList.remove('active'));
+  dots[dotIndex].classList.add('active');
+}
+
+function startUpdateCarousel(carouselSlider) {
+  const dotsContainer = carouselSlider.querySelector('.dots-container');
+  if (!dotsContainer) return; // Exit if dotsContainer doesn't exist
+
+  const dots = dotsContainer.querySelectorAll('.dot');
+  let activeDotIndex = Array.from(dots).findIndex((dot) => dot.classList.contains('active'));
+
+  if (activeDotIndex === -1) {
+    return;
+  }
+
+  const isDesktop = Viewport.isDesktop();
+  let movingForward = true;
+
+  const intervalId = setInterval(() => {
+    if (isDesktop) {
+      if (activeDotIndex === dots.length - 1) {
+        clearInterval(intervalId); // Stop if it's desktop and reaches the last dot
+        return;
+      }
+      activeDotIndex = (activeDotIndex + 1) % dots.length; // Move to the next dot
+    } else {
+      if (activeDotIndex === 0) {
+        movingForward = true; // Switch to moving forward
+      } else if (activeDotIndex === dots.length - 1) {
+        movingForward = false; // Switch to moving in reverse
+      }
+      activeDotIndex = movingForward ? (activeDotIndex + 1) % dots.length : activeDotIndex - 1;
+      if (activeDotIndex < 0) {
+        activeDotIndex = dots.length - 1;
+      }
+    }
+    const activeDot = dots[activeDotIndex];
+    updateCarouselView(activeDot);
+  }, 2000);
+}
+
+function setCarouselView(type, carouselSlider) {
+  const carouselTrack = carouselSlider.querySelector('.carousel-track');
+  const cards = Array.from(carouselTrack.children);
+  const visibleCards = allowedCardsCount();
+  const numberOfDots = cards.length - visibleCards + 1;
+  if (numberOfDots > 1) {
+    const dotsContainer = document.createElement('div');
+    dotsContainer.className = 'dots-container border-box';
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < numberOfDots; i++) {
+      const dot = document.createElement('button');
+      dot.className = 'dot border-box';
+      dot.dataset.index = i;
+      dotsContainer.appendChild(dot);
+      dot.addEventListener('click', function updateCarouselOnClick() {
```

Comment:
```suggestion
      dot.addEventListener('click', () => {
```

Reply:
@sdmcraft since arrow function looses the context of `this`, I am updating this code to 

`dot.addEventListener('click', (event) => {
  updateCarouselView(event.currentTarget);
});`

Reply:
updated